spec.html

```diff
@@ -13524,70 +13526,58 @@ <h1>
+        1. If _strict_ is *false* and _hasParameterExpressions_ is *true*, then
           1. Let _calleeEnv_ be the LexicalEnvironment of _calleeContext_.
-          1. Let _env_ be NewDeclarativeEnvironment(_calleeEnv_).
           1. Assert: The VariableEnvironment of _calleeContext_ is _calleeEnv_.
-          1. Set the LexicalEnvironment of _calleeContext_ to _env_.
+          1. Set the LexicalEnvironment of _calleeContext_ to NewDeclarativeEnvironment(_calleeEnv_).
+        1. Let _env_ be the LexicalEnvironment of _calleeContext_.
         1. For each String _paramName_ of _parameterNames_, do
           1. Let _alreadyDeclared_ be ! _env_.HasBinding(_paramName_).
           1. NOTE: Early errors ensure that duplicate parameter names can only occur in non-strict functions that do not have parameter default values or rest parameters.
           1. If _alreadyDeclared_ is *false*, then
             1. Perform ! _env_.CreateMutableBinding(_paramName_, *false*).
             1. If _hasDuplicates_ is *true*, then
+              1. NOTE: The binding for each unique parameter name is initialized here so that step <emu-xref href="#step-functiondeclarationinstantiation-initialize-duplicate-parameter-bindings"></emu-xref> can invoke IteratorBindingInitialization without an Environment Record (to accommodate the duplicate names).
               1. Perform ! _env_.InitializeBinding(_paramName_, *undefined*).
         1. If _argumentsObjectNeeded_ is *true*, then
-          1. If _strict_ is *true* or if _simpleParameterList_ is *false*, then
-            1. Let _ao_ be CreateUnmappedArgumentsObject(_argumentsList_).
-          1. Else,
-            1. NOTE: A mapped argument object is only provided for non-strict functions that don't have a rest parameter, any parameter default value initializers, or any destructured parameters.
+          1. NOTE: A mapped argument object is only provided for non-strict functions with <emu-xref href="#sec-static-semantics-issimpleparameterlist">simple parameter lists</emu-xref> (which don't contain any parameter initializers, destructured parameters, or rest parameters).
+          1. If _strict_ is *false* and _simpleParameterList_ is *true*, then
             1. Let _ao_ be CreateMappedArgumentsObject(_func_, _formals_, _argumentsList_, _env_).
+          1. Else,
+            1. Let _ao_ be CreateUnmappedArgumentsObject(_argumentsList_).
           1. If _strict_ is *true*, then
             1. Perform ! _env_.CreateImmutableBinding(*"arguments"*, *false*).
-            1. NOTE: In strict mode code early errors prevent attempting to assign to this binding, so its mutability is not observable.
+            1. NOTE: In strict mode code, the immutability of this binding is not observable because early errors prevent attempting to assign to it.
```

Comment:
Introducing the prepositional comma improves flow IMO, but more importantly the use of "immutability" rather than "mutability" clarifies the note (because the binding is in fact immutable, making it strange—even if correct in some senses—to talk about its [lack of] mutability).